docs/source/deprecation.md

```diff
@@ -1,14 +1,164 @@
+utility, such messages are presented inside the generated reports. In case
+of the snactor utility, the information message is printed in the console
+output in the end of the snactor execution. See examples in this page for
+detail.
+
+But be aware of limitations of our detection in leapp and snactor utilities.
+Our detection mechanisms are not bulletproof. Here is list of some examples,
+when our detection does not work well:
+
+- when an actor a message of a deprecated model, the usage of the deprecated
+  model is detected only if the message has been produced before; IOW, in
+  case the message X is produced only on machines with a specific package,
+  the deprecated usage is not detected on machines without the packagw
+- when a message of a deprecated model is consumed, the message points
+  internals of the leapp framework, instead of place where the message is
+  consumed, e.g.
+  ```
+  Usage of deprecated Model "Foo" @ /usr/lib/python2.7/site-packages/leapp/models/__init__.py:113
+  Near:         return cls(init_method='to_model', **data)
+  ```
+- a deprecated class (including model) is reported only when an instance of the
+  class (object) is created during the runtime
+- use of a deprecated function/method is reported only when the function is
+  called
+
+There are probably other cases, but these are the most common, that could
+confuse people.
+
+
+## How to deprecate entities in leapp
+
+When you want to deprecate an entity in leapp projects, use the `deprecated`
+decorator from `leapp.utils.deprecation` above the definition of the entity.
+The decorator has two input parameters to be able to specify the start
+of the deprecation protection period and a message explaining that
+particular deprecation. E.g. in case the deprecated functionality has
+a replacement, it is expected it will be mentioned in the msg.
+See [examples of the use of the @deprecated decorator](deprecation#usage-examples-of-deprecated).
+
+When you mark any entity as deprecated and this entity is then used
+in the code, leapp users
+, user will be notified about that via terminal and report messages.
+As the author of the deprecation, you know that the entity is deprecated
+and you do not want
+to notify people about the code that still uses the deprecated entity
+just for the reason to retain the original functionality. To suppress
+the deprecation messages in such cases,
+use the `suppress_deprecation` decorator taking as arguments objects that
+should not be reported by the deprecation. E.g. in case you use it above the
+definition of an actor, any use of the deprecated entity inside the actor
+will not be reported.
+
+WARNING: It is forbidden to use the `suppress_deprecation` decorator for any
+purposes but one - retaining the original official functionality over the deprecation protection period.
+If you see the message and you are not the *provider* of the functionality, you
+have to update your code to be independent on it.
```

Comment:
I would remove it, because we already specified the use case of the `suppress_deprecation` decorator. Moreover, in my opinion, this sounds a bit rude to write `forbidden` without explaining the reason.

Reply:
I thought the reasoning is clear from the text. If people developers do it,  end user (and even other developers) will not be notified, running the code, so it could happen that something will immediately break after we remove the deprecated functionality. Telling that reasoning in the warning seems for me like duplicate, as it should be really obvious. More importantly, what is suggestion for the correct text?

@bocekm What do you think?